### PR TITLE
Planned fix for security vulnerability for setuptools-65.5.0 lib

### PR DESCRIPTION
**Pull request by bhupendra-vaishnav**
Planned fix for security vulnerability GHSA-r9hx-vwmv-q579 for setuptools-65.5.0 lib

Added or modified files are :
src/python/install.sh
test/python/install_python_setuptools_vulnerability.sh
test/python/scenarios.json

### CODE REVIEW

File: test/python/scenarios.json [hunk 1]
```diff
@@ -1,4 +1,13 @@
 {
+    "install_python_setuptools_vulnerability": {
+        "image": "debian:bookworm",
+        "features": {
+            "python": {
+                "version": "3.10",
+                "installTools": true
+            }
```
Comment (samruddhikhandale) [inline test/python/scenarios.json:8]:
```suggestion
        "image": "python:3:10",
        "features": {
            "python": {
                "version": "none",
                "installTools": true
            }
```

Can we update the scenario to reflect the python image build?

File: test/python/scenarios.json [hunk 2]
```diff
@@ -1,4 +1,13 @@
 {
+    "install_python_setuptools_vulnerability": {
+        "image": "debian:bookworm",
```
Comment (samruddhikhandale) [inline test/python/scenarios.json:3]: Once you address previous comment, can you also add a new test scenario with image `python:3:11` ?

File: src/python/install.sh [hunk 3]
```diff
@@ -454,6 +454,19 @@ if [[ "${INSTALL_PYTHON_TOOLS}" = "true" ]] && [[ $(python --version) != "" ]];
...
+    
+    # Temporary: Removes “setup tools” metadata directory due to https://github.com/advisories/GHSA-r9hx-vwmv-q579
+
+    VULNERABLE_VERSIONS=("3.10" "3.11")
+
+    for vv in "${VULNERABLE_VERSIONS[@]}"; do
+
+        if [ "${PYTHON_VERSION}" == "${vv}" ]; then
+            rm -rf  ${PIPX_HOME}/shared/lib/"python${vv}"/site-packages/setuptools-65.5.0.dist-info
+            #rm -rf ${PYTHON_INSTALL_PATH}/"${VERSION}"/lib/python${vv}/site-packages/setuptools-65.5.0.dist-info
```
Comment (samruddhikhandale) [inline src/python/install.sh:466]: Can you remove this comment?

File: test/python/install_python_setuptools_vulnerability.sh [hunk 4]
```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+set -e
+
+# Optional: Import test library
+source dev-container-features-test-lib
+
+#check "setuptools version 65.5.0 not installed"  bash -c  "ls -lrt /usr/local/py-utils/shared/lib/python3.11/site-packages/setuptools-65.5.0.dist-info"
+#checkPythonPackageVersion "setuptools" "65.5.1"
```
Comment (samruddhikhandale) [inline test/python/install_python_setuptools_vulnerability.sh:9]: Can you add this check?

File: test/python/install_python_setuptools_vulnerability.sh [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+# only for 3.10
+checkVulnerableDir "/usr/local/py-utils/shared/lib/python3.10/site-packages/setuptools-65.5.0.dist-info" "3.10"
+checkVulnerableDir "/usr/local/python/3.10.13/lib/python3.10/site-packages/setuptools-65.5.0.dist-info" "3.10"
+# only for 3.11
+checkVulnerableDir "/usr/local/py-utils/shared/lib/python3.11/site-packages/setuptools-65.5.0.dist-info" "3.11"
+checkVulnerableDir "/usr/local/python/3.11.7/lib/python3.11/site-packages/setuptools-65.5.0.dist-info" "3.11"
+
+# Report result
```
Comment (samruddhikhandale) [inline test/python/install_python_setuptools_vulnerability.sh:67]: Can you also add the following command? this way we can see the status of the files.

```
find . -name "*setuptools*"
```

File: test/python/scenarios.json (hunk 1 above)
Comment (bhupendra-vaishnav) [reply test/python/scenarios.json:8]: These are the points that I wanted to discuss with you. The moment we move to version as "none" the code in install.sh will not work.

File: src/python/install.sh (hunk 3 above)
Comment (bhupendra-vaishnav) [reply src/python/install.sh:466]: I will remove that but I would like to highlight that sometime another directory will get created for setuptools-65.5.0 do do you want us to remove that or not?

File: test/python/scenarios.json (hunk 1 above)
Comment (samruddhikhandale) [reply test/python/scenarios.json:8]: > the code in install.sh will not work.

Which code are you referring at?

File: src/python/install.sh (hunk 3 above)
Comment (samruddhikhandale) [reply src/python/install.sh:466]: > highlight that sometime another directory will get created for setuptools-65.5.0

Do you have a scenario or reproduction for this? Are you aware of the drawbacks of removing that directory? 

We shouldn't add code or comments if it's not 💯 needed. However, if the scanners complain about it, then we can definitely add it back.

File: test/python/scenarios.json (hunk 1 above)
Comment (bhupendra-vaishnav) [reply test/python/scenarios.json:8]: We have create a separate code set with different branch name and created a PR. Its by Gaurav and the details are as below :
python : https://github.com/devcontainers/features/pull/815
debian : https://github.com/devcontainers/features/pull/814

If Debian work in context with Debian image is not require then we can reject  PR # 814.

File: test/python/scenarios.json (hunk 2 above)
Comment (bhupendra-vaishnav) [reply test/python/scenarios.json:3]: Done in PR # 815

File: src/python/install.sh (hunk 3 above)
Comment (bhupendra-vaishnav) [reply src/python/install.sh:466]: I have implemented a way to dynamically fetch the python version so now no issue.

File: test/python/install_python_setuptools_vulnerability.sh (hunk 4 above)
Comment (bhupendra-vaishnav) [reply test/python/install_python_setuptools_vulnerability.sh:9]: Done.

File: test/python/install_python_setuptools_vulnerability.sh (hunk 5 above)
Comment (bhupendra-vaishnav) [reply test/python/install_python_setuptools_vulnerability.sh:67]: Done

**Comment by bhupendra-vaishnav** [comment]
@samruddhikhandale As we have another PR # 814 to address the same issue so should we close/delete this?

**Comment by bhupendra-vaishnav** [comment]
As we have another PR # [814](https://github.com/devcontainers/features/pull/809) to address the same issue so closing this PR.

**Pull request closed without merging**